### ISSUE TITLE
missing attribute error when ActiveRelation.select is used

### ISSUE THREAD

**Issue by wmulligan87** [brainspec/enumerize#211]
I am using ActiveRecord. I have a model (JobLine) using enumerize. The enumerized field (:line_type) has a default set to :stock.

In some methods where I query JobLines, I use the activeRecord 'select' method, which essentially cherry picks attributes from the model, in order to keep it simple (a job_line has 20 or so attributes, in some cases I'm just working with 3 and would prefer to keep away from initializing the other 17 - this is a design decision that has worked well up to now)

Since 1.0, enumerize has thrown a MissingAttribute error when I use this select method, and I'm pretty sure it is related to how the default value is being set on job_line's :after_initialize events. This is because the object that is built, before the after_initialize event, only has the 3 aforementioned 'selected' attributes: it doesn't have the enumerized attribute.

I have come across this behaviour with after_initialize not playing well with ActiveRecord.select and the solution is usually along the lines:

self.enumerized_attribute = xyz if self.has_attribute?(:enumerized_attribute)

So: wondering if anyone else is having this issue, and if this is a valid solution? For now I can just include the enumerized attribute in any select that is made, however this is not a clean or desirable solution moving forward.

**Comment by nashby** [comment]
@wmulligan87 hey, thank you for detailed description! I'm pretty sure we have this case covered in [specs](https://github.com/brainspec/enumerize/blob/master/test/activerecord_test.rb#L97) but probably we're missing something there. Could you please create a simple application that reproduces this issue or create a failing spec? Thanks!

**Comment by nashby** [comment]
Closing this for now. Feel free to ping me if you need any help creating sample application! Thanks.

**Comment by padde** [comment]
@nashby I am having the same issue. I think this is a problem with Rails 3 specifically. Enumerize checks [whether the instance responds to the attribute name](https://github.com/brainspec/enumerize/blob/v1.1.0/lib/enumerize/base.rb#L96), but unfortunately this is not sufficient for Rails 3:

### Rails 4

```ruby
User.select(:id).first.respond_to?(:email)
#=> false

User.select(:id).first.has_attribute?(:name)
#=> false
```

### Rails 3

```ruby
User.select(:id).first.respond_to?(:email)
#=> true

User.select(:id).first.has_attribute?(:name)
#=> false
```

**Comment by padde** [comment]
@wmulligan87 just to confirm this, are you using Rails 3?

**Comment by matsubo** [comment]
I confirmed this happens Rails 4.0.9 with enumerize `0.7.0`.
It resolved by upgrading enumerize to the newest, `1.1.0`.

**Comment by janie624** [comment]
This issue is still present on Rails 3.2.x.
I use enumerize 1.1.1.  
Can't this gem support Rails 3 too?
